integration_tests/src/main/python/hash_aggregate_test.py

```diff
@@ -274,6 +274,7 @@ def test_hash_query_multiple_distincts_with_non_distinct(data_gen, conf):
         'sum(distinct a),' +
         'count(distinct b),' +
         'count(a),' +
+        'count(),' +
```

Comment:
As far as I understand, `count()` needs to be enabled for it to be considered by catalyst. Am I missing something?

Also, for group by, I believe `count()` _shouldn't_ work given the change you made is only for reductions. I tried this manually in my box, and `count()` seemed to be returning 0 for groups of 1, which seemed wrong to me.

If you run out of time today don't worry, I should be able to put up a fix for this.

Reply:
I should've added better tests. The config is only required for Spark3.1+. I have updated the tests now 

>Also, for group by, I believe count() shouldn't work given the change you made is only for reductions. I tried this manually in my box, and count() seemed to be returning 0 for groups of 1, which seemed wrong to me.

Is the CPU result not matching? The unit tests compare the CPU to GPU

Reply:
I re-ran this both for CPU and GPU, and they both return 0 for `count()` even if the groups each would have a positive count.  So I think the `group by` count is ok.

```
scala> spark.sql("select count(),count(*) from foo group by value%2").collect
res8: Array[org.apache.spark.sql.Row] = Array([0,5], [0,5])

scala> spark.conf.set("spark.rapids.sql.enabled", "true")

scala> spark.sql("select count(),count(*) from foo group by value%2").collect
res10: Array[org.apache.spark.sql.Row] = Array([0,5], [0,5])
```

Reply:
Thanks for adding the config.

Reply:
yes, so false will xfail only for Spark3.1+